superset-frontend/plugins/plugin-chart-echarts/src/BigNumber/BigNumberPeriodOverPeriod/buildQuery.ts

```diff
@@ -18,50 +18,38 @@
  */
 import {
   buildQueryContext,
-  getComparisonInfo,
   ComparisonTimeRangeType,
   QueryFormData,
+  getTimeComparisonFiltersByQueryObject,
 } from '@superset-ui/core';
 
 export default function buildQuery(formData: QueryFormData) {
-  const {
-    cols: groupby,
-    time_comparison: timeComparison,
-    extra_form_data: extraFormData,
-  } = formData;
+  const baseQuery = buildQueryContext(formData, baseQueryObject => [
+    baseQueryObject,
+  ]);
 
-  const queryContextA = buildQueryContext(formData, baseQueryObject => [
+  let shiftedFormData = formData;
+  if (
+    formData?.time_comparison === ComparisonTimeRangeType.Custom &&
+    formData?.adhoc_custom
+  ) {
+    shiftedFormData = {
```

Comment:
non-blocking: we've got some more viz plugins with time range comparison coming, so it might be worth it to move this logic to a reusable util. We can do that as a follow up @Antonio-RiveroMartnez

Reply:
Sure, good point.